cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -208,6 +208,22 @@ NUMERIC_FUNCTION(DIVIDE)
 
 #undef DIVIDE
 
+#define POSITIVE(TYPE) \
+  FORCE_INLINE         \
+  gdv_##TYPE positive_##TYPE(gdv_##TYPE in) { return static_cast<gdv_##TYPE>(in); }
+
+NUMERIC_FUNCTION(POSITIVE)
+
+#undef POSITIVE
+
+#define NEGATIVE(TYPE) \
+  FORCE_INLINE         \
+  gdv_##TYPE negative_##TYPE(gdv_##TYPE in) { return static_cast<gdv_##TYPE>(-1 * in); }
```

Comment:
Check [this line](https://github.com/apache/arrow/blob/848fe4e6032f0291e7174cef5019e48ae2a55f21/cpp/src/gandiva/gdv_function_stubs.cc#L776) with the implementation of the ABS function.

You need to take care if the input value is equals to the MIN value for an integer, because when converting to positive you cause loss of digits

Reply:
I think is a good idea to add a test using the `MIN` and `MIN - 1` values to see if it is working correctly.

Reply:
@jpedroantunes Can you address this? -1 * INT_MIN will cause overflow